kotlin/lib/src/main/kotlin/Webhook.kt

```diff
@@ -100,11 +100,20 @@ class Webhook(secret: String) {
         }
     }
 
-    init {
-        var sec = secret
-        if (sec.startsWith(SECRET_PREFIX)) {
-            sec = sec.substring(SECRET_PREFIX.length)
+    constructor(secret: String) {
+        if (secret.startsWith(SECRET_PREFIX)) {
+            secret = secret.substring(SECRET_PREFIX.length)
         }
-        key = Base64.getDecoder().decode(sec)
+        key = Base64.getDecoder().decode(secret)
+    }
+
+    constructor(secret: String, format: String) {
```

Comment:
You can just have a creator function `Webhook.fromRaw()`

Reply:
What does that look like? This is a tricky search to me / Kotlin noob, as it appears there have been multiple approaches through Kotlin's existence. And is this overloaded constructor approach invalid?

Reply:
It's just confusing that it feels very different to everything else we do in all of the languages.

Can't you just define a static function (or maybe a static function on the companion object) that does it?

Reply:
Ok I will take a look at it -- although I feel like this still offers an ergonomic syntax:

```kotlin
Webhook("secret")
Webhook("secret", "raw")
Webhook(bytes)
```

In the meantime, tests are passing on all the other libraries!! Although I don't have runtimes/compilers so they're not verified yet

Reply:
I think using a string for Raw rather than an enum is also not very natural. Actually, I have another thought, will make it elsewhere.